src/spring-cloud/azext_spring_cloud/tanzu.py

```diff
@@ -234,6 +234,10 @@ def tanzu_app_deploy(cmd, client, resource_group, service, name, artifact_path=N
     while result.properties.status == "Building" or result.properties.status == "Queuing":
         sleep(5)
         result = client.build_service.get_build_result(resource_group, service, name, build_result_name)
+    build_logs = client.build_service.get_build_result_log(resource_group, service, name, build_result_name, "all")
+    if build_logs and build_logs.properties and build_logs.properties.blob_url:
+        output = requests.get(build_logs.properties.blob_url).text

```

Comment:
What if the blob didn't completed yet? You can leverage this util to read text out from a block blob: https://github.com/VSChina/azure-cli-extensions/blob/8d78e351d4cc82f350931add8d128a63f5ad5cb7/src/spring-cloud/azext_spring_cloud/_stream_utils.py#L66

Reply:
This will not happen because the API will first make sure all logs are transferred to the blob, and then update the build result to `Succeeded` or `Failed`. It is not streaming (different from standard tier).